File: composeApp/src/commonMain/kotlin/plus/vplan/app/feature/grades/ui/components/detail/components/GivenByRow.kt
package plus.vplan.app.feature.grades.ui.components.detail.components


import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import plus.vplan.app.feature.homework.ui.components.detail.components.MetadataRow
import plus.vplan.app.feature.homework.ui.components.detail.components.MetadataValueContainer
import plus.vplan.app.feature.homework.ui.components.detail.components.tableNameStyle
import plus.vplan.app.feature.homework.ui.components.detail.components.tableValueStyle

@Composable
fun GivenByRow(
    teacherName: String,
) {
    MetadataRow(
        key = {
            Text(
                text = "Erteilt von",
                style = tableNameStyle()
            )
        },
        value = {
            MetadataValueContainer(
                canEdit = false,
                onClick = {}
            ) {
                Text(
                    text = teacherName,
                    style = tableValueStyle(),
                    color = MaterialTheme.colorScheme.onSurfaceVariant,
                    maxLines = 1
                )
            }
        }
    )
}

File: iosApp/iosApp/feature/grades/detail/ui/components/GivenByRow.swift
import SwiftUI

struct GivenByRow: View {
    var teacherName: String

    var body: some View {
        MetadataRow {
            Text("Erteilt von")
                .font(tableNameStyle())
                .foregroundStyle(.gray)
        } value: {
            MetadataValueContainer(canEdit: false, onClick: {}) {
                Text(teacherName)
                    .font(tableValueStyle())
                    .foregroundStyle(Color(.secondaryLabel))
                    .lineLimit(1)
            }
        }
    }
}
